pkg/bee/client.go

```diff
@@ -0,0 +1,538 @@
+package bee
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/beekeeper/pkg/beeclient/api"
+	"github.com/ethersphere/beekeeper/pkg/beeclient/debugapi"
+	bmtlegacy "github.com/ethersphere/bmt/legacy"
+)
+
+// Client manages communication with the Bee node
+type Client struct {
+	api   *api.Client
+	debug *debugapi.Client
+
+	// number of times to retry call
+	retry int
+}
+
+// ClientOptions holds optional parameters for the Client.
+type ClientOptions struct {
+	APIURL              *url.URL
+	APIInsecureTLS      bool
+	DebugAPIURL         *url.URL
+	DebugAPIInsecureTLS bool
+	Retry               int
+}
+
+// NewClient returns Bee client
+func NewClient(opts ClientOptions) (c *Client) {
+	c = &Client{
+		retry: 5,
+	}
+
+	if opts.APIURL != nil {
+		c.api = api.NewClient(opts.APIURL, &api.ClientOptions{HTTPClient: &http.Client{Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: opts.APIInsecureTLS},
+		}}})
+	}
+	if opts.DebugAPIURL != nil {
+		c.debug = debugapi.NewClient(opts.DebugAPIURL, &debugapi.ClientOptions{HTTPClient: &http.Client{Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: opts.DebugAPIInsecureTLS},
+		}}})
+	}
+	if opts.Retry > 0 {
+		c.retry = opts.Retry
+	}
+
+	return
+}
+
+// Addresses represents node's addresses
+type Addresses struct {
+	Overlay   swarm.Address
+	Underlay  []string
+	Ethereum  string
+	PublicKey string
+}
+
+// Addresses returns node's addresses
+func (c *Client) Addresses(ctx context.Context) (resp Addresses, err error) {
+	a, err := c.debug.Node.Addresses(ctx)
+	if err != nil {
+		return Addresses{}, fmt.Errorf("get addresses: %w", err)
+	}
+
+	return Addresses{
+		Ethereum:  a.Ethereum,
+		Overlay:   a.Overlay,
+		PublicKey: a.PublicKey,
+		Underlay:  a.Underlay,
+	}, nil
+}
+
+// Balance represents node's balance with peer
+type Balance struct {
```

Comment:
All Beekeeper packages have their own types, independent from one another.
There was a need for Bee client package from the beginning of Beekeeper project, but this client was never written as part of Bee codebase. Therefore, Package beeclient is written in Beekeeper, with idea that one day it might become part Bee. The rest of Beekeeper code was designed to be completely independent from it, hence type translation.